### PR TITLE
Services: Stub some functions

### PR DESCRIPTION
**Pull request by purpasmart96**
Please note that GenHashConsoleUnique was pretty much taken from 3dmoo

### CODE REVIEW

File: src/core/hle/service/cfg/cfg_u.cpp [hunk 1]
```diff
@@ -47,6 +47,17 @@ static const std::array<u16, 187> country_codes = {
 
 #undef C
 
+/// SecureInfo Region Values
+enum class RegionValue : u32 {
+    JPN = 0,
```
Comment (purpasmart96) [inline src/core/hle/service/cfg/cfg_u.cpp]: Yes I'm of aware that these names don't follow the code guidelines but I can easily change them .

File: src/core/hle/service/cfg/cfg_u.cpp [hunk 2]
```diff
@@ -129,6 +140,61 @@ static void GetConfigInfoBlk2(Service::Interface* self) {
...
+ *  Outputs:
+ *      0 : Result Header code
+ *      1 : Result of function, 0 on success, otherwise error code
+ *      2 : Region value loaded from SecureInfo offset 0x100
+ */
+static void SecureInfoGetRegion(Service::Interface* self) {
+    u32* cmd_buffer = Kernel::GetCommandBuffer();
+
+    cmd_buffer[1] = RESULT_SUCCESS.raw; // No Error
+    cmd_buffer[2] = static_cast<u32>(RegionValue::USA); // Set to USA
```
Comment (archshift) [inline src/core/hle/service/cfg/cfg_u.cpp]: Why not add a config option for this?

File: src/core/hle/service/cfg/cfg_u.cpp [hunk 3]
```diff
@@ -129,6 +140,61 @@ static void GetConfigInfoBlk2(Service::Interface* self) {
...
+ *  Outputs:
+ *      0 : Result Header code
+ *      1 : Result of function, 0 on success, otherwise error code
+ *      2 : Output value
+ */
+static void GetRegionCanadaUSA(Service::Interface* self) {
+    u32* cmd_buffer = Kernel::GetCommandBuffer();
+
+    cmd_buffer[1] = RESULT_SUCCESS.raw; // No Error
+    cmd_buffer[2] = static_cast<u32>(RegionValue::USA); // Set to USA
```
Comment (archshift) [inline src/core/hle/service/cfg/cfg_u.cpp]: Ditto

File: src/core/hle/service/cfg/cfg_u.cpp (hunk 2 above)
Comment (purpasmart96) [inline src/core/hle/service/cfg/cfg_u.cpp]: Well it was kinda out of the scope for the pr, plus I'm afraid  it would of met the same fate as pr #230

Comment (neobrain) [inline src/core/hle/service/cfg/cfg_u.cpp]: It actually makes sense for this one to be exposed in the configuration ;)

**Comment by bunnei** [comment]
Agreed with the others... Please make the region a config option with this PR!

File: src/core/hle/service/cfg/cfg_u.cpp (hunk 3 above)
Comment (Subv) [inline src/core/hle/service/cfg/cfg_u.cpp]: IMO this is semantically misleading, the output value here isn't a region value but 1 or 0.

**Comment by purpasmart96** [comment]
All done, needs reviewing personally now that I look at it, the config could be way better instead of using a integer as a country, I could use a country as a integer. But I found this the simplest to code...

File: src/core/hle/service/cfg/cfg_u.cpp [hunk 4]
```diff
@@ -129,6 +130,62 @@ static void GetConfigInfoBlk2(Service::Interface* self) {
...
+ *      0 : Result Header code
+ *      1 : Result of function, 0 on success, otherwise error code
+ *      2 : Output value
+ */
+static void GetRegionCanadaUSA(Service::Interface* self) {
+    u32* cmd_buffer = Kernel::GetCommandBuffer();
+
+    cmd_buffer[1] = RESULT_SUCCESS.raw; // No Error
+
+    if (Settings::values.region_value = 1) {
```
Comment (Subv) [inline src/core/hle/service/cfg/cfg_u.cpp]: Typo here :P

File: src/core/hle/service/cfg/cfg_u.cpp [hunk 5]
```diff
@@ -129,6 +130,62 @@ static void GetConfigInfoBlk2(Service::Interface* self) {
...
+ *      1 : Result of function, 0 on success, otherwise error code
+ *      2 : Hash/"ID" lower word
+ *      3 : Hash/"ID" upper word
+ */
+static void GenHashConsoleUnique(Service::Interface* self) {
+    u32* cmd_buffer = Kernel::GetCommandBuffer();
+    u32 app_id_salt = cmd_buffer[1];
+
+    cmd_buffer[1] = RESULT_SUCCESS.raw; // No Error
+    cmd_buffer[2] = 0x33646D6F ^ (app_id_salt & 0xFFFFF); // 3dmooSHA
```
Comment (Subv) [inline src/core/hle/service/cfg/cfg_u.cpp]: I _really_ don't like this, it's fine as a stub, but should be marked as so.

**Comment by Subv** [comment]
LGTM after my comments are addressed

File: src/core/hle/service/cfg/cfg_u.cpp (hunk 4 above)
Comment (purpasmart96) [inline src/core/hle/service/cfg/cfg_u.cpp]: Typo where?

Comment (lioncash) [inline src/core/hle/service/cfg/cfg_u.cpp]: The assignment in the conditional

Comment (purpasmart96) [inline src/core/hle/service/cfg/cfg_u.cpp]: @Subv this is the GetRegionCanadaUSA  not the SecureInfoGetRegion, which the enum was for.

File: src/core/hle/service/cfg/cfg_u.cpp (hunk 5 above)
Comment (purpasmart96) [inline src/core/hle/service/cfg/cfg_u.cpp]: Added  a Waring log mentioning that this function is stub

File: src/core/hle/service/cfg/cfg_u.cpp (hunk 4 above)
Comment (Subv) [inline src/core/hle/service/cfg/cfg_u.cpp]: @purpasmart96 I was referring to the 1 in the if condition

Comment (purpasmart96) [inline src/core/hle/service/cfg/cfg_u.cpp]: There's no point of having an enum though.

Comment (Subv) [inline src/core/hle/service/cfg/cfg_u.cpp]: Then the 1 is now a magic number, which is what I wanted to avoid.

Comment (purpasmart96) [inline src/core/hle/service/cfg/cfg_u.cpp]: hmmmm

**Comment by Subv** [comment]
I'd still prefer the enum to be present, but we're discussing over a bikeshed here, it doesn't matter. LGTM

**Comment by purpasmart96** [comment]
@bunnei please review.  :)

**Comment by bunnei** [comment]
LGTM

**Pull request merged**